web/frontend/src/components/base.vue

```diff
@@ -84,14 +105,20 @@ const setDirection = (dir: Directions) => {
   direction.value = dir;
 };
 
-const stop = () => {
+const stop = () => new Promise<void>((resolve) => {
+  console.log('STOPPING');
   stopped = true;
   const req = new baseApi.StopRequest();
   req.setName(props.name);
-  props.client.baseService.stop(req, new grpc.Metadata(), displayError);
-};
+  props.client.baseService.stop(req, new grpc.Metadata(), (error) => {
```

Comment:
Since this is the key bit, can we just make use of the JS event loop by making digest input a promise itself and the .then is based on the resolve here such that queuing happens by way of the dom events firing? That way we avoid our own mini queue

Reply:
Not sure I follow. `digestInput` is a promise now, and the `.then` is based off the resolve in the callback from the `setPower` callback. I guess my question is what would we invoke in the `.then` since we don't really _know_ what the next action is from inside the `digestInput`. We may continue to `digestInput` or if there are no pressed keys then we `stop`. 

I do agree I'd rather not maintain our own event queue here, so anything to reduce that complexity I am open to.